src/AnnotationThread.js

```diff
@@ -115,66 +122,105 @@ class AnnotationThread extends EventEmitter {
         this.state = STATES.inactive;
     }
 
-    /**
-     * Whether or not thread dialog is visible
-     *
-     * @return {void}
-     */
-    isDialogVisible() {
-        return !!(this.dialog && this.dialog.element && !this.dialog.element.classList.contains(CLASS_HIDDEN));
-    }
+    position = () => {};
 
     /**
      * Shows the appropriate annotation dialog for this thread.
      *
+     * @param {Event} event - Mouse event
      * @return {void}
      */
-    showDialog() {
-        // Prevents the annotations dialog from being set up on every call
-        if (!this.dialog.element) {
-            this.dialog.setup(this.annotations, this.element);
+    renderAnnotationPopover(event = null) {
+        if (event) {
+            event.stopPropagation();
+            event.preventDefault();
+        }
+
+        const firstAnnotation = this.annotations[0];
```

Comment:
Why do we need to make a distinction between the first and subsequent annotations?

Reply:
The first annotation is used to determine the position, type, etc. The rest of the annotations are used to just get the comments